.azure-pipelines/test_plan.py

```diff
@@ -178,22 +176,26 @@ def poll(self, test_plan_id, interval=60, timeout=36000):
             status = resp_data.get("status", None)
             result = resp_data.get("result", None)
 
-            if status in ["FINISHED", "CANCELLED", "FAILED"]:
-                if result == "SUCCESS":
-                    print("Test plan is successfully {}. Elapsed {:.0f} seconds" \
-                          .format(status, time.time() - start_time))
+            if status == expected_state or status in ["CANCELLED", "FAILED"]:
```

Comment:
Can the logic be improved to something like below:

* If the current status is expected state, good, return.
* otherwise:
  * If current state in ["CANCELLED", "FAILED", "FINISHED"], there is no point keep waiting, return.
  * otherwise, keep waiting.